### PR TITLE
config: fix validation of OpsGenie configuration

### PR DESCRIPTION
**Pull request by simonpasquier**
The validation should fail if both `api_key` and `api_key_file` are
defined. I think there was a typo in the original PR (#2728) that
enforced `api_url` and `api_key_file` not being defined at the same
time.

cc @jkroepke since you did the initial implementation.

### CODE REVIEW

**Comment by gotjosh** [comment]
Can you rebase from `main`? I think this the failure relates to #2899 which was fixed a few days ago.

**Comment by simonpasquier** [comment]
Thanks! I wanted the change against `release-0.24` to cut v0.24.1 right after since `api_key_file` was added in v0.24.0 :)
I've cherry-picked 35bf59f1829ed998aaf577d56681cf24039fd6a8 instead...

**Comment by gotjosh** [comment]
Silly me! I didn't notice that the target branch was different.

File: config/notifiers.go [hunk 1]
```diff
@@ -492,7 +492,7 @@ func (c *OpsGenieConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 		return err
 	}
 
-	if c.APIURL != nil && len(c.APIKeyFile) > 0 {
+	if c.APIKey != "" && len(c.APIKeyFile) > 0 {
```
Comment (gotjosh) [inline config/notifiers.go:495]:
```suggestion
        if c.APIURL == nil {
          return fmt.Errorf("missing URL in config")
        }
        
	if c.APIKey != "" && len(c.APIKeyFile) > 0 {
```

Comment (gotjosh) [reply config/notifiers.go:495]: I think we need to validate the URL as well, `DefaultOpsGenieConfig` does not contain a default URL.

File: config/notifiers_test.go [hunk 2]
```diff
@@ -590,6 +590,72 @@ func TestOpsgenieTypeMatcher(t *testing.T) {
 	}
 }
 
+func TestOpsGenieConfiguration(t *testing.T) {
```
Comment (gotjosh) [inline config/notifiers_test.go:593]: Should we also include a test here if we decide to include the URL validation?

File: config/notifiers.go (hunk 1 above)
Comment (simonpasquier) [reply config/notifiers.go:495]: there's a default OpsGenie URL in the global config.
https://github.com/prometheus/alertmanager/blob/23f961ec6a9c8a15bf99c83815d35e13d66a4d5d/config/config.go#L591

Comment (gotjosh) [reply config/notifiers.go:495]: 🤦

**Review by gotjosh — APPROVED** [review]
LGTM

**Pull request merged**